driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/ResourceProvider.scala

```diff
@@ -32,14 +32,17 @@ object ResourceProvider {
       }
     }
 
-    private def cacheUri(uri: URI, stream: () => InputStream, message: Path => String): Resource.File = {
+    private def cacheUri(uri: URI, openStream: () => InputStream, message: Path => String): Resource.File = {
       val cachedResource = cached(uri)
       if (!cachedResource.isFile) {
         println(message(cachedResource))
-        Files
-          .createTempFile("cached-resource", "-tmp")
-          .append(stream())
-          .moveTo(cachedResource)
+        val stream = openStream()
+        try {
+          Files
+            .createTempFile("cached-resource", "-tmp")
+            .append(stream)
+            .moveTo(cachedResource)
+        } finally stream.close()
```

Comment:
`append` extension should already close the stream

Reply:
Ah, ok. I will revert. Though a bit confusing. I guess the same function that opens stream should close it.